extensions/ext/GLSL_EXT_spirv_intrinsics.txt

```diff
@@ -0,0 +1,1020 @@
+
+        The `spirv_storage_class` modifies the *StorageClass* operand of
+        declared variables to be the storage class defined for the
+        *StorageClass* enumeration equal to the integer value provided.
+
+        Any extensions or capabilities needed by the storage class must be
+        declared using SPIR-V Requirements Declarations (see section 3.9).
+
+    Add a new section 4.14. SPIR-V Decorations
+
+        SPIR-V decorations not known to or exposed by the GLSL compiler can
+        be added by use of the `spirv_decorate`, `spirv_decorate_id`, or
+        `spirv_decorate_string` qualifiers.
+
+        This functionality is only available when generating SPIR-V.
+
+        The `spirv_decorate`, `spirv_decorate_id`, and
+        `spirv_decorate_string` qualifiers can be used on variable
+        declarations, function parameters, function return types,
+        and expressions (with "constructor" syntax).
+        `spirv_decorate_id` must not be used on structure member
+        declarations, but `spirv_decorate` and `spirv_decorate_string` can.
+
+        spirv_decorate-qualifier:
+
+            `spirv_decorate` ( integer-constant, spirv_decorate-parameter-list )
+            `spirv_decorate` ( spirv-requirements-list , integer-constant, spirv_decorate-parameter-list )
+            `spirv_decorate_id` ( integer-constant, spirv_decorate_id-parameter-list  )
+            `spirv_decorate_id` ( spirv-requirements-list , integer-constant, spirv_decorate_id-parameter-list  )
+            `spirv_decorate_string` ( integer-constant, spirv_decorate_string-parameter-list )
+            `spirv_decorate_string` ( spirv-requirements-list , integer-constant, spirv_decorate_string-parameter-list )
+
+        spirv_decorate-parameter-list:
+
+            spirv_decorate-parameter
+            spirv_decorate-parameter , spirv_decorate-parameter-list
+
+        spirv_decorate-parameter:
+
+            floating-constant
+            integer-constant
+            bool-constant
+
+        spirv_decorate_id-parameter-list:
+
+            constant-expression
+            constant-expression , spirv_decorate_id-parameter-list
+
+        spirv_decorate_string-parameter-list:
+
+            literal-string
+            literal-string , spirv_decorate_string-parameter-list
+
+        The `spirv_decorate` qualifier inserts an *OpDecorate* or
+        *OpMemberDecorate* instruction into the generated SPIR-V code,
+        taking the first integer parameter as its `Decoration` operand,
+        and further parameters as `Extra Operands`.
+        If the `spirv_decorate` qualifier is applied to a struct member,
+        *OpMemberDecorate* is inserted, otherwise *OpDecorate* is.
+        Integer and floating point values used must be expressible in a
+        32-bit range.
+        An error will be generated if the front-end compiler detects a
+        value outside of the 32-bit range.
+
+        The `spirv_decorate_id` qualifier inserts an *OpDecorateId*
```

Comment:
Same comment as for the execution mode ids applies here.

Reply:
OpDecorateId should allow for specialization constants, that was, IIRC, one of the driving reasons behind it. I'll update this.